### PR TITLE
Bump minimum PHP version in composer.json

### PR DESCRIPTION
**Pull request by bramceulemans**

### CODE REVIEW

**Comment by Nyholm** [comment]
is there any technical reason for doing this? Is there something in 7.2+ that we really need?

**Comment by pimjansen** [comment]
> is there any technical reason for doing this? Is there something in 7.2+ that we really need?

Dont think so actually but same goes for php 5 in that matter. Could be an idea to just keep supporting stable php for new releases in my opinion.

**Comment by Nyholm** [comment]
Im always happy to drop PHP versions. But I think we should do it for technical reasons. Just dropping them because they are not supported is not a good idea. We just exclude users. 

There are plenty of other libraries that exclude users in order to "force them to upgrade". 

If 7.1 is causing issues for us or if we really really wanted to use syntax that are only in 7.2+, then I am happy with this PR. 

@bramceulemans: **IF** this PR should get merged, you need to update the CI config in the `.github` directory to not test on PHP 7.1

**Comment by pimjansen** [comment]
I would lets bring it to a clean state now since we removed a lot already and figure out a good migration plan for future use too. 

Those 7.1 users can still use 1.10 which works fine as well.

That said there are more refs in the docs as well that need update then.

**Comment by sanmai** [comment]
If this was a well established project like PHPUnit, I’d suggest to go even further, up to PHP 7.4. But for a hard fork almost nobody knows, well, I suggest we get PHP 5.5 back to the build matrix.

**Comment by bramceulemans** [comment]
My main reason for dropping 7.1 support, is because it's EOL.

I will update the docs and build matrix.

**Comment by Nyholm** [comment]
@bramceulemans I understand that. 
But that is not a good enough reason in my opinion. If it does not give us any benefits and it will exclude users, why would we do such thing?

**Comment by divine** [comment]
Hello,

I really don't know from where this forcing newer PHP version comes from? Is this because of Laravel?

I think there will be (if not already happening) conflict of interests for this library. Is this library for PHP or for only Laravel interests?

PHP EOL doesn't mean it's dead and no updates are pushed, actually it's being taken care by upstream providers such as Redhat https://access.redhat.com/support/policy/updates/rhscl#life_cycle

Basically, please stop forcing when not required only because it's EOL. Some projects can't be updated instantly like you did it with one line here.

Thanks!

**Comment by Nyholm** [comment]
I appreciate you making a PR. But since there is no win merging this, then I'll close this.
